src/main/java/org/example/controllers/UserController.java

```diff
@@ -15,4 +35,70 @@ private String showMainPage() {
     public String showAdminPage() {
         return "/adminpage";
     }
+
+    @GetMapping("/usererror")
+    public String error() {
+        return "/usererror";
+    }
+
+    @GetMapping("/usersuccess")
+    public String successfully() {
+        return "/usersuccess";
+    }
+
+    @GetMapping("/registration")
+    public String createUser(ModelMap modelMap) {
+        modelMap.addAttribute("user", new UserDTO());
+        return "/registration";
+    }
+
+    @PostMapping("/registration")
+    public String saveUser(UserDTO userDTO) {
+        if (userService.checkIfUserExistsByLogin(userDTO.getLogin())) {
+            return "/usererror";
+        } else {
+            userService.saveUser(userDTO);
+            return "redirect:/login";
+        }
+    }
+
+    @GetMapping("/useredit")
+    public String editUser(@RequestParam("userId") int id, ModelMap modelMap) {
+        modelMap.addAttribute("user", new UserDTO());
+        modelMap.put("user", userService.findById(id));
+        return "/useredit";
+    }
+
+    @PostMapping("/useredit")
+    public String saveEditUser(@RequestParam("userId") int id, UserDTO userDTO) {
+        userService.updateUser(userDTO, userService.findById(id));
```

Comment:
`updateUser1 method shoul have userId as a second parameter instead of user, and `userService.findById(id)` should be called inside it